internal/bminventory/inventory.go

```diff
@@ -2787,31 +2790,33 @@ func (b *bareMetalInventory) calculateHostNetworks(log logrus.FieldLogger, clust
 func (b *bareMetalInventory) listClustersInternal(ctx context.Context, params installer.V2ListClustersParams) ([]*models.Cluster, error) {
 	log := logutil.FromContext(ctx, b.log)
 	db := b.db
+
+	var dbClusters []*common.Cluster
+	var clusters []*models.Cluster
+
 	if swag.BoolValue(params.GetUnregisteredClusters) {
 		if !identity.IsAdmin(ctx) {
 			return nil, common.NewApiError(http.StatusForbidden, errors.New("only admin users are allowed to get unregistered clusters"))
 		}
 		db = db.Unscoped()
 	}
-	var dbClusters []*common.Cluster
-	var clusters []*models.Cluster
-	whereCondition := make([]string, 0)
 
-	filterByOrg := b.authHandler.EnableOrgTenancy()
-	if user := identity.AddOwnerFilter(ctx, "", filterByOrg, swag.StringValue(params.Owner)); user != "" {
-		whereCondition = append(whereCondition, user)
+	db = b.authzHandler.OwnedBy(ctx, db)
+
+	if params.Owner != nil {
+		db = db.Where("user_name = ?", params.Owner)
 	}
```

Comment:
This is probably worth a comment.
Just reading these lines makes it feel like we don't need to include the additional `params.Owner` bit.
But in this case, `OwnedBy` is not actually looking for a specific owner, right?

Reply:
it is. We had a lengthy discussion about this implementation today with @avishayt. I will summarize it in a comment

Reply:
Maybe put another way, it _may_ not be looking for a specific owner.
I'm guessing the reason this exists is that because even if tenancy is enabled we might only want to see clusters owned by a specific user, right?

I would definitely add a comment or maybe add another paramter to `OwnedBy` to indicate that we want to filter by a specific user even if tenancy is enabled.